### PR TITLE
Avoid unnecessary warnings when loading `CLIPConfig`

### LINKED ISSUE #28042: Confusing deprecation / warning messages
### System Info

```
- `transformers` version: 4.37.0.dev0
- Platform: Linux-5.15.0-88-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.19.3
- Safetensors version: 0.4.1
- Accelerate version: 0.24.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.1.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

### Who can help?

@amyeroberts for Vision

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

When doing:

```py
from diffusers import StableDiffusionPipeline

pipe = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
```

I'm getting a couple of confusing error messages since transformers 4.36:

```
`AnnotionFormat` is deprecated and will be removed in v4.38. Please use `transformers.image_utils.AnnotationFormat` instead
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["bos_token_id"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["eos_token_id"]` will be overriden.
```

None of these variables (`AnnotionFormat` or `text_config_dict`) are defined anywhere in `diffusers` or in the configs: https://huggingface.co/runwayml/stable-diffusion-v1-5/blob/main/text_encoder/config.json

It seems like something inside Transformers triggers these deprecation warnings which makes the messages very confusing and non-actionable for users. Also since it happens every time `from_pretrained(...)` is called, it clutters the CLI quite a bit

### Expected behavior

No warnings or clearer instructions and what needs to be changed to remove these warnings

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Avoid unnecessary warnings when loading `CLIPConfig`: when a user doesn't change something inside `text_config`.

Fix #28042

### CODE REVIEW

File: src/transformers/models/clip/configuration_clip.py
Comment (ydshieh) [inline src/transformers/models/clip/configuration_clip.py:335]: This is for `id2label`

File: src/transformers/models/clip/configuration_clip.py
Comment (ydshieh) [reply src/transformers/models/clip/configuration_clip.py:350]: IMO, this is not just `info` : if a user changes  `text_config`.

If a user doesn't change anything --> it could be info. 

So no matter if we want to use `info` or `warning`, I would have to use the above conditions to separate the cases.

Comment (ydshieh) [reply src/transformers/models/clip/configuration_clip.py:350]: I can keep the condition, and if it holds, give info rather than skip them. Is this OK?

Comment (ydshieh) [reply src/transformers/models/clip/configuration_clip.py:351]: new logic is added.

File: src/transformers/models/clip/configuration_clip.py
Comment (patrickvonplaten) [reply src/transformers/models/clip/configuration_clip.py:335]: Sorry could you explain this in a bit more detail. Why do we need to take `"id2label"` into account here?

Comment (ydshieh) [reply src/transformers/models/clip/configuration_clip.py:335]: - `id2label` in the line `_text_config_dict = CLIPTextConfig(**text_config_dict).to_dict()`:
  - `CLIPTextConfig(**text_config_dict)` is a config object, and its `id2label` has `int` (`0`, `1`, etc) as keys, see [here](https://github.com/huggingface/transformers/blob/71d47f0ad498b7649f11d3a9cca3cd3585e4341f/src/transformers/configuration_utils.py#L332)
  - `to_dict` will keep `id2label` untouched: still has `int` (`0`, `1`, etc) as keys

- `text_config` argument is loaded from the `text_config` attribute in a config json file --> it is a dictionary (not an config object) at this sage, and its `id2label` is also a dictionary with `str` (`"0"`, `"1"`) as keys.

Comment (ydshieh) [reply src/transformers/models/clip/configuration_clip.py:335]: we have to make them having the same format, otherwise their `id2label` are different and the warning would be given

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this so quickly! 

I don't think there's an easy solution to this, unfortunately. Let a comment, LMKWYT

**Comment by ydshieh** [comment]
>Hard-coding values like this is brittle

This is fine IMO for this situation

> The intention of the user is ambiguous. If I deliberately set "bos_token_id": 0 in the config - then I still would have unexpected behaviour but the warning wouldn't show.

Very much agreed for this!

Since we don't have a perfect solution to this issue caused by the original design, and 2 core maintainers accept change it to `info` --> Let's do this. (At least, we have this handling block, so it's easy to figure thing out when someone get trapped by [this issue](https://github.com/huggingface/transformers/pull/19954#issuecomment-1295182328))

**Comment by ydshieh** [comment]
> Do we know if users ever modify the log verbosity for debugging purposes? i.e. would it be enough to set everything to info?

The ratio of this is quite low I believe.

**Comment by ydshieh** [comment]
Changed to `info`. There are are config classes in the clip family like `altclip`, `chineseclip` etc. Will change them too before merge unless there is other opinions.

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and iterating to find a solution for this! 

Even though other clip-releated models haven't caused users to complain about the warnings, I think it'd be better for them to be aligned with CLIP here

**Pull request merged**
